src/commponents/MarvelHome.tsx

```diff
@@ -1,100 +1,85 @@
-    await this.props.fetchHeroSaga(this.props.location.search);
-  }
+type IProps = RouteComponentProps<{ location: string }>;
 
-  onSearchChange(e: React.ChangeEvent<HTMLInputElement>): void {
-    this.setState({ searchField: e.target.value });
-  }
+const MarvelHome: React.FC<IProps> = (props: IProps) => {
+  const { heroes, loading, error } = useTypeSelector((state) => state.heroes);
+  const { fetchHero } = useActions();
+  const history = useHistory();
+  const location = useLocation();
+  // const [searchField, setSearchField] = useState<string>();
+  const [heroesPerPage] = useState<number>(8);
+  // const [nameUrl, setNameUrl] = useState<string | string[]>();
+  const parser = queryString.parse(history.location.search.substr(1));
+  const actualName = parser.name;
 
-  filterHeroes(posts: IHero[]): IHero[] {
-    const searchField = this.state.searchField.toLowerCase();
-    return posts.filter((hero) => hero.name.toLowerCase().includes(searchField));
-  }
+  useEffect(() => {
+    fetchHero(0, heroesPerPage, actualName || null);
+    console.log('mount');
+  }, []);
 
-  paginationHandleChange(event: React.ChangeEvent<unknown>, page: number): void {
-    this.setState({ currentPage: page });
-  }
+  useEffect(() => {
+    if (history.action === 'POP' || location.pathname === '/') {
+      fetchHero(0, heroesPerPage, actualName || null);
+    }
+    console.log('update');
+  }, [location]);
 
-  checkOnPosts(currentPosts: IHero[]): IHero[] {
-    if (this.state.searchField === '') {
-      return this.filterHeroes(currentPosts);
+  const onSearchChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    const name = e.target.value;
+    if (name) {
+      history.push(`/?name=${name}`);
+      fetchHero(0, heroesPerPage, name);
     } else {
-      return this.filterHeroes(this.props.heroes);
+      history.push('');
+      fetchHero(0, heroesPerPage, null);
     }
-  }
+  };
+
+  const paginationHandleChange = (event: React.ChangeEvent<unknown>, page: number): void => {
+    const offset = heroesPerPage * page - heroesPerPage;
+    fetchHero(offset, heroesPerPage, null);
+  };
 
-  render(): React.ReactNode {
-    const indexOfLastPost = this.state.currentPage * this.state.heroesPerPage;
-    const indexOfFirstPost = indexOfLastPost - this.state.heroesPerPage;
-    const currentPosts = this.props.heroes.slice(indexOfFirstPost, indexOfLastPost);
-    const filterHeroes = this.checkOnPosts(currentPosts);
-    return (
-      <div>
-        <Header />
-        {!this.props.loading ? (
-          <>
-            <Search onSearchChange={this.onSearchChange} />
-            <HeroList heroes={filterHeroes} />
-            <Paginations
-              heroesPerPage={this.state.heroesPerPage}
-              totalPosts={this.props.heroes.length}
-              handleChange={this.paginationHandleChange}
-            />
-          </>
-        ) : (
+  const errorHandlerHeroes = () => {
+    if (!loading && !error) {
```

Comment:
better to write if(error) { return ... }, if(loading) { return... },  { return... } to reduce logic